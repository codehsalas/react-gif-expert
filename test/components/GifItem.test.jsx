import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en GifItem", () => {

  const title = 'Saitama';
  const url   = 'http://one-punch.com/saitama.jpg';

  test('Debe hacer match con el SnapShot', () => {
    const { container } = render( <GifItem title={ title } url={ url } /> )
    expect( container ).toMatchSnapshot();
  });

  test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
    render( <GifItem title={ title } url={ url }/> );
    //screen.debug();
    //expect( screen.getByRole('img').src).toBe( url );
    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( url );
    expect( alt ).toBe( alt );
  });


  test('Verificar si se muestra el titulo', () => {
    render( <GifItem title={ title } url={ url } />);
    expect( screen.getByText( title )).toBeTruthy();
  });

});