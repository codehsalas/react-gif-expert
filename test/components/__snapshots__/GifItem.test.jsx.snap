// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el SnapShot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
